### ISSUE TITLE
TypeError exception on Redis DB layer

### ISSUE THREAD

**Issue by kulicuu** [rfcx/rfcx-worker-analysis#23]
>Hi Wylie. Can you look into the Redis DB layer either today or tomorrow? The CLI is crashing due to a TypeError exception. Thanks!

**Comment by kulicuu** [comment]
Steps to reproduce the error ?

**Comment by flygeneticist** [comment]
Ran the CLI with the following arguments:
python analyze.py -wav ~/path/toTestAudio.wav -data
'sampleMetadata.properties'

CLI run up until the anomaly_detection module / redis-db module. Here's the
stack trace:
*Traceback (most recent call last):*
*  File "analyze.py", line 93, in <module>*
*    main()*
*  File "analyze.py", line 89, in main*
*    analyzer.analyze(sound)*
*  File
"/home/kak9699/Documents/code/RFCx/rfcx-worker-analysis/modules/service_layer.py",
line 60, in analyze*
*    anomaly_detection.AnomalyDetector(self.logger,
repo).determine_anomaly(prof_final)*
*  File
"/home/kak9699/Documents/code/RFCx/rfcx-worker-analysis/modules/domain_modules/anomaly_detection.py",
line 23, in determine_anomaly*
*    model = self.repo.get_model(guardian_id)*
*  File
"/home/kak9699/Documents/code/RFCx/rfcx-worker-analysis/modules/db_layer.py",
line 55, in get_model*
*    sumSquareDifZ = stationH['sumSquareDifZ']*
*KeyError: 'sumSquareDifZ'*

**Comment by kulicuu** [comment]
cool cool; i'm going  to look at this now.

**Comment by kulicuu** [comment]
Can I use a/the m4a file as sample or need to use a wav file maybe ?  

I'm getting

```
wylie ~/southernOcean/RainForestCxx/rfcx-worker-analysis $ python analyze.py -wav ~/1420614653101.m4a -data sampleMetaData.properties 
2015-01-11 12:43:05,014 - services - INFO - Reading sound file 60819a58-69e9-4253-87bf-eb462aa846b2
2015-01-11 12:43:05,015 - services - ERROR - Read-in failed for file: 60819a58-69e9-4253-87bf-eb462aa846b2
	global name 'self' is not defined
```

**Comment by kulicuu** [comment]
Maybe need a good sample wav file ?  One avail for download ?

**Comment by flygeneticist** [comment]
Just need to convert the m4a to a wav file first.

**Comment by kulicuu** [comment]
Ah cool will do that then.

**Comment by kulicuu** [comment]
```
wylie ~/southernOcean/RainForestCxx/rfcx-worker-analysis $ python analyze.py -wav ~/sample.wav -data sampleMetaData.properties 
2015-01-11 13:01:20,225 - services - INFO - Reading sound file 60819a58-69e9-4253-87bf-eb462aa846b2
2015-01-11 13:01:20,267 - services - INFO - Read-in successful for file: 60819a58-69e9-4253-87bf-eb462aa846b2
/home/wylie/southernOcean/RainForestCxx/rfcx-worker-analysis/modules/domain_modules/spectral_analysis.py:50: RuntimeWarning: divide by zero encountered in log10
  self.db_arr = 20.*np.log10(self.abs_arr)
2015-01-11 13:01:21,030 - services - INFO - Completed spectral analyis for file: 60819a58-69e9-4253-87bf-eb462aa846b2
2015-01-11 13:01:21,043 - services - INFO - Completed fingerprinting for file: 60819a58-69e9-4253-87bf-eb462aa846b2
/home/wylie/southernOcean/RainForestCxx/rfcx-worker-analysis/modules/domain_modules/sound_classification.py:20: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if profile.peaks != None:
2015-01-11 13:01:21,044 - services - INFO - Classifed file as having sound(s): ['chainsaw', 'car', 'bird']
2015-01-11 13:01:21,044 - services - INFO - Completed classification for file: 60819a58-69e9-4253-87bf-eb462aa846b2
Traceback (most recent call last):
  File "analyze.py", line 93, in <module>
    main()
  File "analyze.py", line 89, in main
    analyzer.analyze(sound)
  File "/home/wylie/southernOcean/RainForestCxx/rfcx-worker-analysis/modules/service_layer.py", line 60, in analyze
    anomaly_detection.AnomalyDetector(self.logger, repo).determine_anomaly(prof_final)
  File "/home/wylie/southernOcean/RainForestCxx/rfcx-worker-analysis/modules/domain_modules/anomaly_detection.py", line 33, in determine_anomaly
    model.train(column)
AttributeError: 'NoneType' object has no attribute 'train'

```

**Comment by flygeneticist** [comment]
Wylie, your error and stack trace is what I got as well after flushing the redis DB with the "flushall" command.

**Comment by kulicuu** [comment]
Okay so where is this attribute 'train' supposed to come from ?

**Comment by flygeneticist** [comment]
Here's how I followed the module logic to find the change needed:
In line 23, model is assigned the model in the Redis database, based on the
guardian_id.
This model lookup didn't find anything, and returned None.
This None was then asked access "train" which it didn't have and threw the
error.

This all came about due to the test for model, on line 25, checking for it
not being equal to "no such station!". This was from an earlier Redis
setup. After I changed it to check for the model != None, the program
correctly creates the new model from the Gaussian class.
